src/Controller/IndexController.php

```diff
@@ -162,25 +167,26 @@ public function getFieldsAction(Request $request, EventDispatcherInterface $even
         $eventDispatcher->dispatch($event, AdminEvents::GET_INDEX_FIELD_NAMES_PRE_SEND_DATA);
         $data = $event->getArgument('data');
 
-        return $this->adminJson(['data' => array_values($data)]);
+        return $this->jsonResponse(['data' => array_values($data)]);
     }
 
     /**
      * @Route("/get-all-tenants", name="pimcore_ecommerceframework_index_getalltenants", methods={"GET"})
      *
-     * @return \Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse
+     * @param TranslatorInterface $translator
+     * @return JsonResponse
      */
-    public function getAllTenantsAction(): \Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse
+    public function getAllTenantsAction(TranslatorInterface $translator): JsonResponse
     {
         $tenants = Factory::getInstance()->getAllTenants();
         $data = [];
 
         if ($tenants) {
             foreach ($tenants as $tenant) {
-                $data[] = ['key' => $tenant, 'name' => $this->trans($tenant)];
+                $data[] = ['key' => $tenant, 'name' => $translator->trans($tenant)];
```

Comment:
```suggestion
                $data[] = ['key' => $tenant, 'name' => $translator->trans($tenant, [], 'admin')];
```
Otherwise it tried to fetch the translation from messages catalog.

Reply:
how do I know when to use what?

Reply:
All the controller extending from `AdminController` before just used `$this->trans` [which passed the params](https://github.com/pimcore/admin-ui-classic-bundle/pull/29/files#diff-15f7f071121bfa3ffa4bf6f5b31db091f12a2ce65acae1642ab42c45e0c9eef7L281) so now need to pass explicitly.